### PR TITLE
test: Add list tests to RTE

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
Increase test coverage of the Kaizen RTE component

## What
- add list tests

### CODE REVIEW

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+      const decreaseIndentBtn = screen.getByRole("button", {
+        name: "Decrease indent",
+      })
+      const increaseIndentBtn = screen.getByRole("button", {
+        name: "Increase indent",
+      })
+
+      expect(decreaseIndentBtn).toHaveAttribute("aria-disabled", "true")
+      expect(increaseIndentBtn).toHaveAttribute("aria-disabled", "true")
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:151]: Is there a `toBeDisabled` assertion that can be used here?

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      fireEvent.focus(screen.getByRole("textbox", { name: "List RTE" }), {
+        target: { textContent: "this will be a ul" },
+      })
+
+      await waitFor(() => {
+        userEvent.click(screen.getByRole("button", { name: "Bullet List" }))
+        screen.getByRole("list")
+
+        expect(screen.getByRole("list")).toBeInTheDocument()
+        expect(document.querySelectorAll("ul").length).toBeGreaterThan(0)
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:83]: Is the second assertion here redundant? Wouldn't the first one fail if there's no ul?

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    ]
+
+    it("indent buttons are 'disabled' by default", () => {
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:140]: /nit these test cases don't read nicely like your others "it indent buttons..."

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx (hunk 2 above)
Comment (mcwinter07) [reply packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:83]: That is true, hang up from when I was tryign to be thorough to verify it was working, the less assertions the better hey

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx (hunk 1 above)
Comment (mcwinter07) [reply packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:151]: there is, but sadly it doesn't capture aria-disabled, only disabled={true}

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx (hunk 3 above)
Comment (mcwinter07) [reply packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx:140]: nit away :) could definitely update this!

**Review by dougmacknz — APPROVED** [review]

**Pull request merged**